components/financial_assistance/app/domain/financial_assistance/operations/applications/aptc_csr_credit_eligibilities/renewals/resubmit.rb

```diff
@@ -0,0 +1,129 @@
+# frozen_string_literal: true
+
+require 'dry/monads'
+require 'dry/monads/do'
+
+module FinancialAssistance
+  module Operations
+    module Applications
+      module AptcCsrCreditEligibilities
+        module Renewals
+          # This Operation is used to resubmit renewal draft applications that remain in a 'renewal_draft' or 'submitted' state
+          # ::FinancialAssistance::Operations::Applications::AptcCsrCreditEligibilities::Renewals::Resubmit.new.call({renewal_year: 2025})
+          class Resubmit
+            include Dry::Monads[:result, :do]
+
+            # @param [Hash] opts The options to resubmit renewal_draft applications
+            # @option opts [Integer] :renewal_year (required)
+            # @return [Dry::Monads::Result]
+            def call(params)
+              validated_params      = yield validate(params)
+              renewal_eligible_apps = yield applications_for_families(validated_params)
+              resubmission_results  = yield resubmit_applications(renewal_eligible_apps)
+              Success(resubmission_results)
+            end
+
+            private
+
+            # Validates the parameters passed to the operation.
+            #
+            # @param params [Hash] The parameters to validate.
+            # @option params [Integer] :renewal_year The renewal year.
+            # @return [Success, Failure] Returns a Success object containig the params if validation passes, otherwise returns a Failure object with an error message.
+            def validate(params)
+              return Failure('Missing renewal_year key') unless params.key?(:renewal_year)
+              return Failure("Invalid value: #{params[:renewal_year]} for key renewal_year, must be an Integer") if params[:renewal_year].nil? || !params[:renewal_year].is_a?(Integer)
+              Success(params)
+            end
+
+            # Fetches the current year applications that are eligible for renewal in the given renewal year.
+            # The current year is assumed to be the year before the given renewal year.
+            #
+            # @param params [Hash] The parameters to filter applications.
+            # @option params [Integer] :renewal_year The renewal year to filter applications.
+            # @return [Success, Failure] Returns a Success object with eligible applications if found, otherwise returns a Failure object with an error message.
+            def applications_for_families(params, family_ids = ::HbxEnrollment.individual_market.enrolled.current_year.distinct(:family_id))
```

Comment:
Only this command we can use to fetch application to resubmit them. we can ignore all the other queries and if conditions in this operation.
```
FinancialAssistance::Application.no_timeout.by_year(@renewal_year).where(:family_id.in => renewal_eligible_family_ids, :aasm_state.in => ["submitted", "renewal_draft"])
```